File: 9-11/solutions/sol_c.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cstdint>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;
    string s;
    cin >> s;
    vector<int> letters_count('z' - 'a' + 1);
    for (int i = 0; i < n; ++i)
        ++letters_count[s[i] - 'a'];

    string full;
    for (char i = 'a'; i <= 'z'; ++i) {
        if (letters_count[i - 'a'])
            full += i;
    }
    for (int i = 0; i < full.size(); ++i)
        --letters_count[s[i] - 'a'];

    int result = INT32_MAX;
    for (int i = (int) full.size(); i < n; ++i) {
        bool enough_letters = true;
        int count_operations = 0;
        vector<int> letters_copy = letters_count;
        for (int j = 0; j < full.size(); ++j) {
            if (s[i - full.size() + j] != full[j]) {
                ++count_operations;
                if (--letters_copy[full[j] - 'a'] < 0) {
                    enough_letters = false;
                    break;
                }
            }
        }
        if (enough_letters)
            result = min(result, count_operations);
    }

    if (result == INT32_MAX)
        cout << -1;
    else
        cout << result;

    return 0;
}

File: 9-11/solutions/sol_c.py
n = int(input())
s = input()
letters_count = [0 for _ in range(ord('z') - ord('a') + 1)]
for i in range(n):
    letters_count[ord(s[i]) - ord('a')] += 1

full = ""
for i in range(len(letters_count)):
    if letters_count[i]:
        full += chr(ord('a') + i)
for i in range(len(full)):
    letters_count[ord(s[i]) - ord('a')] -= 1

result = 10 ** 10
for i in range(len(full), n):
    enough_letters = True
    count_operations = 0
    letters_copy = letters_count.copy()
    for j in range(len(full)):
        if s[i - len(full) + j] != full[j]:
            count_operations += 1
            letters_copy[ord(full[j]) - ord('a')] -= 1
            if letters_copy[ord(full[j]) - ord('a')] < 0:
                enough_letters = False
                break
    if enough_letters:
        result = min(result, count_operations)

if result == 10 ** 10:
    print(-1)
else:
    print(result)